libpal/libpal.h

```diff
@@ -34,19 +44,33 @@ typedef struct pal_cpuid_register_values {
     uint32_t edx;
 } pal_cpuid_register_values;
 
-#ifdef __cplusplus
-extern "C"
-#endif
-pal_cpuid_register_values pal_execute_cpuid(uint32_t eax, uint32_t ecx);
+__extern pal_cpuid_register_values pal_execute_cpuid(uint32_t eax, uint32_t ecx);
 
-#ifdef __cplusplus
-extern "C"
-#endif
-uint64_t pal_execute_rdmsr(uintptr_t ecx);
+__extern uint64_t pal_execute_rdmsr(uintptr_t ecx);
 
-#ifdef __cplusplus
-extern "C"
+__extern uint32_t pal_execute_vmread(uintptr_t addr);
+
+__extern uint64_t pal_execute_cr0_read();
+
+#ifdef _M_X64 /* Microsoft-specific macro set when targeting x64 */
+/**
+ * MASM for x64 will give an A2007 (undefined symbol) error
+ * for assembly refers to cr1 (presumably because cr1 is
+ * reserved).
+ *
+ * In this case just return 0.
+*/
+uint64_t pal_execute_cr1_read() { return 0; }
```

Comment:
I'm guessing you included access mechanisms for CR1 because there's a definition for that control register in the project's data directory. Instead of adding these, I think I'd rather a) delete that data definition or b) keep the data definition but delete the "access_mechanisms" section.

Reply:
Okay, IMO I think deleting the "access_mechanisms" piece would make the most sense.